### ISSUE TITLE
bug: S3 requests 404 after v1.2, X-Amz-Security-Token now matters

### ISSUE THREAD

**Issue by mrocke-moxe** [localstack/localstack#7084]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Current Behavior

S3 requests that include the header `X-Amz-Security-Token` now return a 404 with LocalStack v1.2 where in v1.1 this header seems to have been ignored.

This may apply to other services beyond S3--I only was able to test S3 given my use case.

### Expected Behavior

When a request is made that includes the `X-Amz-Security-Token` header it is ignored OR there is some way to tell LocalStack to ignore this header.

### How are you starting LocalStack?

With a `docker run` command

#### How are you starting localstack (e.g., `bin/localstack` command, arguments, or `docker-compose.yml`)

Dockerfile:
```
FROM localstack/localstack:latest

ENV SERVICES=sqs,s3,dynamodb
ENV DEBUG=true
ENV DEFAULT_REGION=us-east-1
```

Command line:
```
docker build --pull --no-cache . -t MyApp-localstack
docker run --rm -it -p 4566:4566 MyApp-localstack
```

`.aws` contains a default profile with a real AWS aws_access_key_id and aws_secret_access_key

#### Client commands (e.g., AWS SDK code snippet, or sequence of "awslocal" commands)

Example 1:
```
aws s3 ls --endpoint-url=http://localhost:4566
```
The above command returns a 404 in v1.2 where in v1.1 it did not (listed buckets instead).

Example 2: If a second profile is created in `.aws/config` as follows:
```
[profile localstack]
aws_access_key_id = test
aws_secret_access_key = test
```
this command will succeed:
```
aws s3 ls --endpoint-url=http://localhost:4566 --profile=localstack
```

In Example 2, the http header `X-Amz-Security-Token` is not present in the request.  In example 1 this header is present.  I looked at the headers in v1.1 and v1.2 and in v1.1 the `X-Amz-Security-Token` header is present in the request for Example 1 and returns the expected results (list of buckets).

### Environment

```markdown
- OS: Windows 11
- LocalStack: latest, 1.2
```

### Anything else?

I asked about this on the community slack and it was recommended that I open an issue for investigation.  I am not clear if this is a true bug or is part of a new feature/expected behavior to consider the `X-Amz-Security-Token` header.  In my use case we have several apps that need real AWS services for things like Secrets Manager but also need the LocalStack services like S3.  Updating our apps to use two different profiles is easier said than done unfortunately - so, I'm looking for a solution on the LocalStack side.

Slack post:  https://localstack-community.slack.com/archives/CMAFN2KSP/p1666287124244789

**Comment by steffyP** [comment]
Thank you for creating the issue @mrocke-moxe!

I suspect this is related to issue #6869 (which was closed by the reporter because the workaround was acceptable).

**Comment by bentsku** [comment]
Hi @mrocke-moxe and thanks for your report! We've introduced some changes regarding the multi-account access, and it's possible your buckets are now accessible between accounts. Could you pull the `latest` version and give it a try? Thanks a lot!

**Comment by bentsku** [comment]
I'll close the issue as it should be fixed thanks to #8395, feel free to reopen if you still encounter the issue!
